internal/db/repo.go

```diff
@@ -1427,6 +1427,24 @@ func GetRepositoriesByForkID(forkID int64) ([]*Repository, error) {
 	return getRepositoriesByForkID(x, forkID)
 }
 
+func getRepositoriesNonMirror(e Engine) ([]*Repository, error) {
+	repos := make([]*Repository, 0, 10)
+	err := e.Where("is_mirror = ?", false).Find(&repos)
+	if err != nil {
+		return repos, err
+	}
+	for _, r := range repos {
+		owner, _ := GetUserByID(r.OwnerID)
+		r.Owner = owner
+	}
```

Comment:
This is too magical, we should use `r.FullName()` inside the loop in the `backup.go`.